view/src/extras/asynchronous_input.js

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import TextField from "@material-ui/core/TextField";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import axios from "axios";
+import { authMiddleWare } from "../util/auth";
+
+/**
+ * Creates an asynchronous input that will query database for options
+ * when it is opened by the user (to avoid loading on page reload). 
+ * @param props Propositions passed from parent that determine where the 
```

Comment:
I always thought props was short for properties...

Reply:
they definitely are! I'll change this tomorrow morn